rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -149,10 +148,21 @@ def _load_vocabulary_params(self) -> None:
             if self.OOV_words:
                 self.OOV_words = [w.lower() for w in self.OOV_words]
 
-        # Additional vocabulary size to be kept reserved
-        self.additional_vocabulary_size = self.component_config[
-            "additional_vocabulary_size"
+        additional_size_attributes = [
+            key
+            for key, value in self.component_config[
+                "additional_vocabulary_size"
+            ].items()
+            if value
         ]
+        if additional_size_attributes:
+            rasa.shared.utils.io.raise_deprecation_warning(
+                f"additional vocabulary size has been specified for attributes - "
+                f"{additional_size_attributes}. The parameter `additional_vocabulary_size` "
+                f"has been deprecated since the pipeline does not create an extra buffer for "
+                f"vocabulary anymore. Any value assigned to this parameter will be ignored. "
+                f"You can omit specifying `additional_vocabulary_size` in future runs."
```

Comment:
```suggestion
                f"The parameter `additional_vocabulary_size` has been specified for attributes - "
                f"`{additional_size_attributes}`. The parameter  "
                f"has been deprecated since the pipeline no longer creates an extra buffer for "
                f"additional vocabulary. Any value assigned to this parameter will be ignored. "
                f"You can omit specifying `additional_vocabulary_size` in future runs."
```